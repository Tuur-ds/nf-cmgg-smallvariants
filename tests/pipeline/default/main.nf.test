nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_default"

    options "-stub"

    test("pipeline_default - default") {

        when {
            params {
                outdir = "${outputDir}"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("default") }
            )
        }

    }

}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
