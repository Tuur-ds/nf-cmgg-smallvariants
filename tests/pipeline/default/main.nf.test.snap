{
    "pipeline_default - default": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.bed",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.ped",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.html",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T13:55:37.532795185"
    }
}